### ISSUE TITLE
Bad explanation on Waypoint: Use Hex Code for Specific Colors

### ISSUE THREAD

**Issue by afduarte** [FreeCodeCamp/FreeCodeCamp#5392]
Challenge [Waypoint: Use Hex Code for Specific Colors](http://www.freecodecamp.com/challenges/waypoint-use-hex-code-for-specific-colors#?solution=%3Cstyle%3E%0A%20%20body%20%7B%0A%20%20%20%20background-color%3A%20%23000000%3B%0A%20%20%7D%0A%3C%2Fstyle%3E%0A) has an issue.
User Agent is: <code>Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/47.0.2526.80 Safari/537.36</code>.
Please describe how to reproduce this issue, and include links to screenshots if possible.

My code:

Doesn't matter

Hexadecimal has nothing to do with the fact that you use 6 characters for representing the colour... That's because you use 2 digits (8 bit) for every component of the colour (RGB). If you we had hypothetically hexadecimal representation of a CMYK colour scheme, it would use 8 digits and you would still call it hexadecimal.

I don't think you even need to explain what's hexadecimal, that's out of the scope of the course...

I like your effort, but if you want to give extra information, it should be the right one.

Nonetheless, good course, I'm enjoying it.

**Comment by raisedadead** [comment]
Hi,
 
While I understood the point you are making here, But still IMO, I think the explanation about using Hexadecimal scheme for representing the RGB is apt, in this case.

Its true that the word "hexadecimal" has nothing to do with representing color, where we are using 6 digits for it and each digit is a hexadecimal.

But having said, that in CSS it has been common to call the color code as "hex". Its kind of an coding/programming slang.

For example : If someone asks a CSS developer, 'Hey! Whats the hex code for red?' He would reply, 'Oh! its # FF0000', He would understand what's being asked, even though it may not be logical to connect the word "hex" and color, as you mentioned, and to which I agree too.

But, yet and still, it becomes essential that students are acquainted with the common terms, so they are comfortable with such terms, which they will face every where.

Secondly, the brief intro is also essential over here about hexadecimal, because of the fact that 6 digits are a standard way of representing the 16 Million color combinations that can be accommodated. And people who have no knowledge of how these decimals-hexadecimals work, need some place to start with. Hence, there's a link to more detailed explanation at Wikipedia, too.

Again, The point of the course is to target any one without any prior knowledge of coding to get started with coding. So, I guess its alright to put what's in the context.

What do you say?

**Comment by SaintPeter** [comment]
@raisedadead is correct here.  Hexadecimal is commonly used in programming and some users will never have encountered it before.  We offer up an explanation in line and in context.

I've never actually had anyone argue for REMOVING content from FCC before, but there ya go.

Closing this as we will not be reevaluating our pedagogical decisions on this waypoint.

**Comment by afduarte** [comment]
@raisedadead completely missed my point. What I meant is that you try to explain what is headecimal but end up saying it is called hexadecimal because you use 6 digits to represent a color (#123456) and because it uses 6 letters (ABCDEF), which is not true! Hexadecimal means a base 16 number system, has nothing to do with 6! Although hexa does mean 6 most of the times, here it means 16! I just think it is misleading to tell someone that it is called hexadecimal because there are 6 digits in the hexadecimal code color! also, further on, you end up telling people it might just be 3, not 6... I see someone asking themselves: "Why is it called hexadecimal when there's 3?!?"

You do what you want to do with this, It's not bikeshedding, I'm happy to fork this, fix the problem myself and submit a pull request, but it makes me unhappy that a website focused on learning doesn't care about the text you create for people that are learning.

My goal is not to remove content from FCC, I just want to give my opinion on how to improve it. If you want to take it, you take it, if you won't care, I won't care trying to improve it any more, I just thought the whole point of having a community based learning platform was for the community to give feedback and improve the website content as a community.

I wonder if you actually understand what I mean here and if you gave it some thought before dismissing it as "bikeshedding"

**Comment by SaintPeter** [comment]
What text would you propose?

**Comment by afduarte** [comment]
I would suggest this:

>Did you know there are other ways to represent colors in CSS? 
>One of these ways is called hexadecimal code, or hex code for short.
>You may be familiar with decimal being a number like 57 or 0.333. In CSS, decimal can be thought of >as the numbers zero through nine, giving a total of ten numbers.
>Hexadecimal (or hex) is a numeral system with a base of 16. This means it uses sixteen distinct >symbols.
>As with decimal, the symbols 0–9 represent values zero to nine, and A, B, C, D, E, F represent values >ten to fifteen.
>Putting these together, each place value of a hexadecimal can be represented by 0 to F, giving us a >total of 16 possible values. You can find more information about hexadecimal numbers here (link to >wikipedia)

>With CSS, we use 6 hexadecimal numbers to represent colors, two for each  component, red, green >and blue. You can find more information about the RGB color system here. For example, #000000 is >the lowest possible value, and it represents the color black.

>Replace the word black in our body element's background-color with its hex code representation, >#000000.

You can see it doesn't look bad on the page by the attached image.

I just think it is a more accurate explanation for someone that has never heard about hexadecimal before. It doesn't go into much detail, so it won't be boring and won't put people off (which was your goal in the first place, I completely understand that), but won't be inaccurate, so people interested in learning more won't have conflicting ideas when researching about hexadecimal.

The fact is that people don't need to know what is hexadecimal or to understand it in order to use it in CSS (which was my original point when you understood I wanted it removed), but since you decided to explain it (which I completely agree with), I think it should be done in an accurate way. I hope you don't think I was being arrogant or anything, I really like the website and the project, I just thought I could help improving this Waypoint.

![freecodecamp](https://cloud.githubusercontent.com/assets/9434822/11919492/6caab8be-a74c-11e5-9485-824661aff112.png)

**Comment by raisedadead** [comment]
Yup, now its much clearer what you were saying @afduarte ! And, Haha its funny now, But I am sorry your original argument, did lead us to think otherwise.

Lets see what @SaintPeter and others have to say about this , but I am cool with the change you propose.

Cheers, and sorry about the confusion.
Thanks! :+1:

**Comment by SaintPeter** [comment]
I made some minor edits:
> Did you know there are other ways to represent colors in CSS? 
One of these ways is called hexadecimal code, or hex code for short.
We most commonly use decimal, or base 10 numbers, which use the symbols 0 through 9 for each digit.
Hexadecimal (or hex) is a numeral system with a base of 16. This means it uses sixteen distinct symbols.
As with decimal, the symbols 0–9 represent values zero to nine, and A, B, C, D, E, F represent values ten to fifteen.
Putting these together, each place value of a hexadecimal can be represented by 0 to F, giving us a total of 16 possible values. You can find more information about hexadecimal numbers here (link to wikipedia)

> With CSS, we can use 6 hexadecimal numbers to represent colors, two for each component, red, green and blue. You can find more information about the RGB color system here. For example, #000000 is the lowest possible value, and it represents the color black.

> Replace the word black in our body element's background-color with its hex code representation, #000000.

I'm concerned that this is still a bit verbose - probably one of the longest descriptive sections we have - which it was to begin with, but this is even longer.  I don't know if it's possible to streamline it any more than it already is.  

I'm willing to discuss this, though.

**Comment by erictleung** [comment]
Here's my go at making the text less verbose and as concise as possible. (Bold text indicates my changes from @SaintPeter's suggestions.)

> Did you know there are other ways to represent colors in CSS?  One of these ways is called hexadecimal code, or hex code for short.
>
> We **usually** use decimal**s**, or base 10 numbers, which use the symbols 0 **to** 9 for each digit. Hexadecimal**s** (or hex) **are base 16 numbers**. This means it uses sixteen distinct symbols. **Like** decimals, the symbols 0–9 represent values zero to nine. **Then** A, B, C, D, E, F represent values ten to fifteen. **Altogether, 0 to F can represent a digit in hexadecimals**, giving us 16 **total** possible values. You can find more information about hexadecimal numbers here.
>
> **In** CSS, we can use 6 hexadecimal **digits** to represent colors, two **each** for **the** red **(R)**, green **(G)** and blue **(B)** components. For example, #000000 **is black** and is **also** the lowest possible value. You can find more information about the RGB color system here. 
>
> Replace the word black in our body element's background-color with its hex code representation, #000000.

In the second to last paragraph, I also switched around the sentences to make it flow a little better.

The live version is 167 words with 756 characters (no spaces). My version is 159 words with 776 characters (no spaces). @SaintPeter's changes have 175 words and 830 characters (no spaces).

**Comment by SaintPeter** [comment]
This looks really good.  Pretty concise, too.

Minor question:
>  Altogether, 0 to F can represent a digit in hexadecimal**s**,

Shouldn't `hexadecimal` be singular in this case?  We're referring to the numbering system, not the digits.

Otherwise, this looks good to go.

**Comment by erictleung** [comment]
Good point. I'll keep it singular.

I've put in a PR and tried to highlight appropriately. Let me know if there needs to be any additions and removal of highlighting.

Also, I've added the `target='blank'` attribute to the links because I noticed there's that tiny figure with a square and arrow pointing out next to the links. To me that implies that what I'm going to click on will open a new tab. If that's not what you intended, I can remove the target attribute.
